pkg/outputs/file/file_output.go

```diff
@@ -142,6 +149,8 @@ func (f *File) Init(ctx context.Context, name string, cfg map[string]interface{}
 		f.file = os.Stdout
 	case "stderr":
 		f.file = os.Stderr
+	case "rotating":
+		f.file = newRotatingFile(f.cfg)
```

Comment:
I wonder if it makes more sense to define `Rotation` as a pointer and handle it in the `default:` case.
If `cfg.Rotation != nil`, then create a rotating file (after setting the missing fields to their defaults).

Reply:
I think that makes a lot more sense, and makes the config a bit easier to understand. Updated!